heartbeat/monitors/wrappers/summarizer/summarizer.go

```diff
@@ -0,0 +1,152 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package summarizer
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/gofrs/uuid"
+
+	"github.com/elastic/beats/v7/heartbeat/eventext"
+	"github.com/elastic/beats/v7/heartbeat/monitors/jobs"
+	"github.com/elastic/beats/v7/heartbeat/monitors/stdfields"
+	"github.com/elastic/beats/v7/heartbeat/monitors/wrappers/monitorstate"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+type Summarizer struct {
+	rootJob        jobs.Job
+	contsRemaining uint16
+	mtx            *sync.Mutex
+	jobSummary     *JobSummary
+	checkGroup     string
+	stateTracker   *monitorstate.Tracker
+	sf             stdfields.StdMonitorFields
+}
+
+type JobSummary struct {
+	Attempt      uint16                   `json:"attempt"`
+	MaxAttempts  uint16                   `json:"max_attempts"`
```

Comment:
For any `max_attempts` value > 2, I only seem to get one retry on the UI. Tested with the following:
```yaml
- type: http
  enabled: true
  max_attempts: 5
  id: http_down_3
  name: http_down_3
  urls: ["http://localhost:9200"]
  schedule: '@every 1m'
```